File: examps/Queue/Queue.java
/*
TorXakis - Model Based Testing
Copyright (c) 2015-2017 TNO and Radboud University
See LICENSE at root directory of this repository.
*/


/*********************************************************************
 Class for Queue
 *********************************************************************/

class Queue
{  private Cell first = null;
   private Cell last  = null;

   public void add(int x)
   {  Cell h = new Cell();
      h.data = x;
      h.rest = null;
      if (empty())
      {  first = h; }
      else
      {  last.rest = h; } ;
      last = h;
   }

   public boolean empty()
   {  return(first==null); }

   public int take()
   {  Cell h = first;
      first = h.rest;
      return(h.data);
   }

   public void show()
   {  System.out.print("[ ");
      Cell h = first;
      while (h != null)
      {  System.out.print(h.data);
         h = h.rest;
         if (h != null)
         {  System.out.print(", ");
         }
      }
      System.out.println(" ]");
   }
      
}

/*********************************************************************/

File: test/sqatt/src/ExploreModels/Queue.hs
{-
TorXakis - Model Based Testing
Copyright (c) 2015-2017 TNO and Radboud University
See LICENSE at root directory of this repository.
-}

{-# LANGUAGE OverloadedStrings #-}
module ExploreModels.Queue (exampleSet) where

import           Examples.Paths
import           Prelude        hiding (FilePath)
import           Sqatt

exampDir :: FilePath
exampDir = "Queue"

testModel :: TxsExample
testModel = emptyExample
  { exampleName = "Stepper Test 200"
  , txsModelFiles = [txsFilePath exampDir "Queue"]
  , txsCmdsFiles = [txsCmdPath exampDir "Queue_Stepper_Model"]
  , expectedResult = Pass
  }

testLossyModel :: TxsExample
testLossyModel = emptyExample
  { exampleName = "Stepper Test (Lossy) 200"
  , txsModelFiles = [txsFilePath exampDir "Queue"]
  , txsCmdsFiles = [txsCmdPath exampDir "Queue_Lossy_Stepper_Model"]
  , expectedResult = Pass
  }

examples :: [TxsExample]
examples = [testModel, testLossyModel]

exampleSet :: TxsExampleSet
exampleSet = TxsExampleSet "Queue #model" examples
